core/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -176,24 +187,6 @@ public RecoveryResponse recoverToTarget() throws IOException {
                 throw new RecoveryEngineException(shard.shardId(), 1, "prepare target for translog failed", e);
             }
 
-            // engine was just started at the end of phase1
```

Comment:
correct, the reason for having this check here was to make sure that recovery of a replica cannot complete once a primary is relocated. We now have this check again in the finalizeRecovery stage (when we call `runUnderOperationPermit(() -> shard.markAllocationIdAsInSync`) so we're covered.